packages/angular/ssr/src/i18n.ts

```diff
@@ -43,3 +43,158 @@ export function getPotentialLocaleIdFromUrl(url: URL, basePath: string): string
   // Extract the potential locale id.
   return pathname.slice(start, end);
 }
+
+/**
+ * Parses the `Accept-Language` header and returns a list of locale preferences with their respective quality values.
+ *
+ * The `Accept-Language` header is typically a comma-separated list of locales, with optional quality values
+ * in the form of `q=<value>`. If no quality value is specified, a default quality of `1` is assumed.
+ * Special case: if the header is `*`, it returns the default locale with a quality of `1`.
+ *
+ * @param header - The value of the `Accept-Language` header, typically a comma-separated list of locales
+ *                  with optional quality values (e.g., `en-US;q=0.8,fr-FR;q=0.9`). If the header is `*`,
+ *                  it represents a wildcard for any language, returning the default locale.
+ *
+ * @returns A `ReadonlyMap` where the key is the locale (e.g., `en-US`, `fr-FR`), and the value is
+ *          the associated quality value (a number between 0 and 1). If no quality value is provided,
+ *          a default of `1` is used.
+ *
+ * @example
+ * ```js
+ * parseLanguageHeader('en-US;q=0.8,fr-FR;q=0.9')
+ * // returns new Map([['en-US', 0.8], ['fr-FR', 0.9]])
+
+ * parseLanguageHeader('*')
+ * // returns new Map([['*', 1]])
+ * ```
+ */
+function parseLanguageHeader(header: string): ReadonlyMap<string, number> {
```

Comment:
Question: One other thing which occurred to me is that many headers can be repeated and have semantics for how to handle them. What happens if a request does:

```
Accept-Language: en
Accept-Language: fr;q=0.5, it;q=0.4
Accept-Language: *
```

Is there any specced behavior for how to handle multiple `Accept-Language` headers?

Reply:
Those will be merged into a single string. 

See: https://developer.mozilla.org/en-US/docs/Web/API/Headers/get#examples

Reply:
Oh nice, didn't realize that was built-in to the `Request` / `Response` types. I recommend having at least one test which covers this case since it's something devs tend to overlook about HTTP headers.